### PR TITLE
[ports/rp2] new machine_rtc.c module

### PR DESCRIPTION
**Pull request by astrorafael**
Hi
Although I was delighted to see micropython support in the new Raspberry Pi Pico, I as a bit surprised that there was no `machine.RTC` class for this port. So I took the plunge and began writting a basic `machine_rtc.c` module and learning in the process a bit about how to make an extension C module. This module is very basic for the time being (no interrupt handling) and conforms (I hope) to the interface documented in  the [MicroPython 1.14 official documentation]( https://docs.micropython.org/en/latest/library/machine.RTC.html?highlight=rtc#machine.RTC).

The python test code is this:
```python
from machine import RTC
from utime import sleep, sleep_ms

def test_main():
    rtc = RTC(0)
    print(rtc)
    print(rtc.now())
    remaining = rtc.alarm(0, 5000, repeat=False)
    while (remaining >0):
        print("{0:d} ms. left".format(remaining))
        sleep(1)
        remaining = rtc.alarm_left()
    print("done")

def test_cancel():
    rtc = RTC(0)
    print(rtc)
    print(rtc.now())
    remaining = rtc.alarm(0, 5000, repeat=False)
    while (remaining >0):
        print("{0:d} ms. left".format(remaining))
        sleep(1)
        remaining = rtc.alarm_left()
        rtc.cancel()

def test_1sec():
    rtc = RTC(0)
    rtc.init((2015, 1, 1, 1, 0, 0, 0, None))
    sleep_ms(1002)
    print(rtc.now())
```

### CODE REVIEW

**Comment by astrorafael** [comment]
Ok. I see that I have to get throught the formatting code & commit workflow. Will resubmit when ready.

**Comment by stinos** [comment]
Looks ok on first sight, but can you start by fixing style inconsistencies and other issues: squash in everything into logical commits (in this case: 2 of them, one which modifies .gitignore because that is unrelated to the rest, the other one which ads the module). Then, some things are pretty hard to read: we don't generally use hard-to-maintain mixed comment styles like `// --------------------` or `//######` or '/**'. Instead just use one single comment style with `// `. Except for the license (which should also look like all others i.e. not start with your custom text).

File: ports/rp2/machine_rtc.c [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+
+// ----------------------
+// Pico specific includes
+// ----------------------
+
+#include "hardware/rtc.h"
+#include "pico/util/datetime.h"
+
+// ------------------
+// Exported constants
```
Comment (stinos) [inline ports/rp2/machine_rtc.c:47]: There's no such thing as 'exporting' a PP define

File: ports/rp2/machine_rtc.c [hunk 2]
```diff
@@ -0,0 +1,457 @@
...
+
+/* 
+ * This should be defined and exported in the Python machine module
+ * but apparently it is not used
+ */
+#define RP2_PWR_MODE_IDLE         (0x01)
+#define RP2_PWR_MODE_SLEEP        (0x02)
+#define RP2_PWR_MODE_DEEPSLEEP    (0x04)
+
+enum dotw {Sunday = 0, Monday, Tuesday, Wednesday, Thursday, Friday, Saturday};
```
Comment (stinos) [inline ports/rp2/machine_rtc.c:64]: Can you spell this out for readability?

Comment (astrorafael) [reply ports/rp2/machine_rtc.c:64]: Do you mean writing a comment for dotw (day of the week) ?

Comment (stinos) [reply ports/rp2/machine_rtc.c:64]: Rather just writing day_of_the_week or something like that: immediately clear and no comment needed

Comment (astrorafael) [reply ports/rp2/machine_rtc.c:64]: ok

File: ports/rp2/machine_rtc.c (hunk 1 above)
Comment (astrorafael) [reply ports/rp2/machine_rtc.c:47]: ok

Comment (astrorafael) [reply ports/rp2/machine_rtc.c:47]: Regarding the number of commits, this means that I'll have to delete my fork and start from scratch, right?
Also I have to fix the commit message style.

Comment (stinos) [reply ports/rp2/machine_rtc.c]: > this means that I'll have to delete my fork and start from scratch

No. There are a bunch of ways to do this. Given the number of commits: something like `git reset --mixed` will keep all your changes but erase all commits, so you can create a single new one.

Comment (astrorafael) [reply ports/rp2/machine_rtc.c]: Finally I started from scratch and issued PR  #6929 . I tried git reset --mixed but wasn't sure about the results and didn't want to mess it up with the commits.
In the mean time, it seems another developer wants to PR (#6928) another implementation of the same port and functionality. 
Funny ...

Comment (astrorafael) [reply ports/rp2/machine_rtc.c]: t seems that, somehow, updating my local & remote repos with upstream and doing git reset --mixed, made PR #6929 to close. I did not want to close it on purpose. Now there is a #6932, with all green lights. Sorry for all this mess and thank you for your support. It is also ok with me if the project chooses #6928.

**Comment by astrorafael** [comment]
Closed, since it is duplicated with #6932 and I know now how to make PRs complying with MicroPython standards

**Pull request closed without merging**